Add command line reference guide to the Tutorials page

calekochenour opened the pull request:
Closes #202 

This PR adds a reference guide for the commands used throughout the package creation process. The commands are grouped into four categories:

- Environment Setup
- Package Development
- Package Publishing
- Miscellaneous Commands

Adds a new section to the [Tutorials](https://www.pyopensci.org/python-package-guide/tutorials/intro.html) page and a corresponding grid to the [landing](https://www.pyopensci.org/python-package-guide/index.html) page. See screenshots for a preview from my local docs build.

![01-command-line-reference-section](https://github.com/pyOpenSci/python-package-guide/assets/54423680/8fb3f024-b0df-4252-83c0-272e017afe2b)

![02-command-line-reference-grid](https://github.com/pyOpenSci/python-package-guide/assets/54423680/1796021b-fafd-428e-a086-55fd471a503b)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206982522639506

lwasser commented:
@calekochenour thank you for this!! just a note that i have seen it and i'll put it on my todo list to review in the upcoming week!!

lwasser reviewed (dismissed):
nice work on this @calekochenour i'm going to share this in our slack with the community! you are also welcome to join our slack if you want. email me if that is of interest leah at pyopensci.org

JacksonBurns commented on tutorials/command-line-reference.md:
```diff
@@ -0,0 +1,85 @@
+| Description | Syntax |
+|---|---|
+| Create package structure and baseline contents | `hatch new [PACKAGE_NAME]` |
+| Install package locally in editable mode | `python -m pip install -e .` |
```
I don't think this is in scope for the PR but I thought I might mention it in case it is - when pip installing optional packages like `pandas[test]`, you have to quote the `[]` on some shells (like powershell), i.e. `pip install -e ".[test]"`, since it's a reserved character.

lwasser replied:
oh yes - we actually discuss the quotes [here](https://www.pyopensci.org/python-package-guide/package-structure-code/declare-dependencies.html#install-dependency-groups) in the packaging guide. you could add a small breakout @calekochenour about this and then link to that page.

lwasser replied:
we could also do this. @calekochenour to make this more easily mergable - we can address the issues here adding a windows specific section and merge it. and then open an issue to add a linux / mac section for those commands so you don't have to add all of the os specific stuff in one per! that might be best.

calekochenour replied:
@lwasser That works. I can address the windows-specific issues and then have a separate issue/PR for linux and mac updates.

lwasser replied:
perfect! yes i think that is best. then we can get this pr merged and you or anyone can always add a section for mac / linux later!

calekochenour replied:
Added an entry for installing dev dependencies using the ".[GROUP]" notation.

willingc replied:
Thanks!

JacksonBurns commented on tutorials/command-line-reference.md:
```diff
@@ -0,0 +1,85 @@
+
+| Description | Syntax |
+|---|---|
+| Set PowerShell execution policy | `Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser` |
```
Windows user here! Yes, these are the commands to install on Windows, identical to those on Scoop's website: https://scoop.sh/

May be best to link to their site and simply say 'install scoop'

calekochenour replied:
Added (Windows) to the descprition of OS-specific commands. Also included a note in the "Important" section at the beginning indicating that there are Windows-specific commands and that mac/linux will be added in the future.

calekochenour commented:
@lwasser Thanks for the review and comments. I will address them soonest. I did test this on Windows, so some of the commands are OS-specific. The example commands at the moment reflect what I actually tested, except for the upload to Conda-forge. Happy to add more for various operating systems, pip vs. pipx, as well as your other comments.

calekochenour commented:
@JacksonBurns Thanks for your comments as well. I will address these soon.

willingc commented on tutorials/command-line-reference.md:
```diff
@@ -0,0 +1,85 @@
+
+| Description | Syntax |
+|---|---|
+| View environments hatch has access to | `hatch env show` |
```
```suggestion
| View environments | `hatch env show` |
```

calekochenour replied:
Updated to say "View environments".

willingc commented on tutorials/command-line-reference.md:
```diff
@@ -0,0 +1,85 @@
+
+:::
+
+## Miscellaneous Commands
```
Let's rename this section to Versions and Environments (or some alternative). Miscellaneous is a term that is difficult for a new user to unpack.

calekochenour replied:
Updated section name to "Versions and Environments".

willingc reviewed (commented):
Looking good. A few suggestions on one section.

lwasser commented:
hey friends! just checking in. what do we need to do to get this merged?

i thijnk from my perspective @calekochenour we can just be super specific about the windows commands (telling users this is for windows) and then we can go back in a separate PR and add mac/linux commands later! if that is the case then i think the changes would be mostly focused on headers or section text that specify "windows" and then maybe a note that says we will be updating this with mac/linux commands in the future. let me know what you think! 
Let's keep the changes simple so we can get this merged!

calekochenour commented:
@lwasser Thanks for the ping on this. I have not forgotten. I will look to make the updates/changes soonest. What you suggested in your latest comment makes sense to me.

calekochenour commented on tutorials/command-line-reference.md:
```diff
@@ -0,0 +1,85 @@
+| Set PowerShell execution policy | `Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser` |
+| Install Scoop | `Invoke-RestMethod -Uri https://get.scoop.sh \| Invoke-Expression` |
+| Add "main" bucket as download source | `scoop bucket add main` |
+| Add "versions" bucket as download source | `scoop bucket add versions` |
```
Same comment as above. Latest commit addresses this. Added (Windows) to the description of OS-specific commands. Also included a note in the "Important" section at the beginning indicating that there are Windows-specific commands and that mac/linux will be added in the future.

calekochenour commented on tutorials/command-line-reference.md:
```diff
@@ -0,0 +1,85 @@
+| Install Scoop | `Invoke-RestMethod -Uri https://get.scoop.sh \| Invoke-Expression` |
+| Add "main" bucket as download source | `scoop bucket add main` |
+| Add "versions" bucket as download source | `scoop bucket add versions` |
+| Install pipx | `scoop install pipx` or `scoop install main/pipx` |
```
This will be addressed when the mac/linux specific commands are added.

calekochenour commented on tutorials/command-line-reference.md:
```diff
@@ -0,0 +1,85 @@
+| Add "versions" bucket as download source | `scoop bucket add versions` |
+| Install pipx | `scoop install pipx` or `scoop install main/pipx` |
+| Install python | `scoop install python` or `scoop install main/python` |
+| Install specific python version | `scoop install versions/python311` |
```
Removed in latest commit. I like the idea of a breakout/separate section for python install. Can save this for another update.

calekochenour commented on tutorials/command-line-reference.md:
```diff
@@ -0,0 +1,85 @@
+| List hatch commands | `hatch -h` |
+| Open location of hatch config file | `hatch config explore` |
+| Print contents of hatch config file | `hatch config show` |
+| Install grayskull | `pipx install grayskull` |
```
Added `pip` install for `hatch` and `grayskull`.

calekochenour commented:
@lwasser @JacksonBurns @willingc I have addressed all comments with my latest commits. Please let me know if there are any outstanding issues and/or any comments that I have missed. Thanks in advance.

JacksonBurns reviewed (commented):
All looks great! Thanks for taking my notes into account.

willingc commented:
I'll need to troubleshoot why this is not building. I tried to fix the merge conflict.

willingc commented:
I just reran CI. There is one warning still present related to a toctree caption: https://github.com/pyOpenSci/python-package-guide/actions/runs/10276079239/job/28435889487?pr=211#step:9:74

Thanks for the updates @calekochenour. Please reach out if you need guidance on fixing this warning. 👍🏼

willingc commented on tutorials/intro.md:
```diff
@@ -50,6 +50,14 @@ Add a license & code of conduct <add-license-coc>
 
+:::{toctree}
+:hidden:
+
```
```suggestion
```

willingc commented:
@lwasser @JacksonBurns If you can approve this PR, we can merge as the CI errors are now resolved :smile:

JacksonBurns reviewed (commented):
It won't let me press approve, but I do approve haha

JacksonBurns commented:
@lwasser thanks! Just accepted.
